test/core/gridmapping/test_dataset.py

```diff
@@ -30,6 +30,23 @@ def test_from_regular_cube(self):
         self.assertEqual((2, 180, 360), gm.xy_coords.shape)
         self.assertEqual(('coord', 'lat', 'lon'), gm.xy_coords.dims)
 
+    def test_from_regular_cube_with_crs(self):
+        dataset = xcube.core.new.new_cube(variables=dict(rad=0.5),
+                                          x_start=0,
+                                          y_start=0,
+                                          x_name='x',
+                                          y_name='y',
+                                          crs='epsg:25832')
```

Comment:
Test does not work if you also pass correct units (m). I mention this also in my comment on new.py

Reply:
See my response there.